### PR TITLE
cmake configuration cleanup

### PR DESCRIPTION
**Pull request by christophgysin**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -4,39 +4,46 @@ project(sway C)
 
 set(FALLBACK_CONFIG_DIR "/etc/sway" CACHE PATH
   "Fallback config directory defaults to /etc/sway")
-add_definitions('-DFALLBACK_CONFIG_DIR=\"${FALLBACK_CONFIG_DIR}\"')
+add_definitions(-DFALLBACK_CONFIG_DIR=\"${FALLBACK_CONFIG_DIR}\")
 
-set(CMAKE_C_FLAGS "-g")
+if("${CMAKE_BUILD_TYPE}" STREQUAL "")
+	set(CMAKE_BUILD_TYPE Debug)
+endif()
```
Comment (SirCmpwn) [inline CMakeLists.txt]: I'd like to force debug information so that we get sane backtraces after crashes in the log.

**Comment by SirCmpwn** [comment]
Looks good save for that one comment.

**Comment by christophgysin** [comment]
Removed the offending commit.

**Comment by SirCmpwn** [comment]
Thanks. Will merge once Travis is good.

**Comment by SirCmpwn** [comment]
Travis is FUBAR now, they made the cutover to a system that does not work with our build requirements. I'll build this locally.

**Pull request merged**